### PR TITLE
[DOC] embedding function documentation

### LINKED ISSUE #30: Add documentation on embedding functions
- how to create an embedding function
- how to integrate with:
- openai
- cohere
- huggingface
- any custom model

### PR DESCRIPTION
**Pull request by changhiskhan**
closes #30

### CODE REVIEW

File: docs/src/embedding.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+Using an embedding function, you can apply it to raw data
+to generate embeddings for each row.
+
+Say if you have a pandas DataFrame with a `text` column that you want to be embedded,
+you can use the following code to generate embeddings and add create a combined
+pyarrow table:
+
+```python
+from lancedb.embeddings import with_embeddings
```
Comment (eddyxu) [inline docs/src/embedding.md:58]: add a link to `with_embedding` python API?

File: docs/src/embedding.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Embedding Functions
+
+Embeddings are high dimensional floating-point vector representations of your data or query.
+Anything can be embedded using some embedding model or function.
+For a given embedding function, the output will always have the same number of dimensions.
+
+## Creating an embedding function
+
+Any function that takes as input a batch (list) of data and outputs a batch (list) of embeddings
```
Comment (eddyxu) [inline docs/src/embedding.md:9]: is input always the same length as outputs?

File: docs/src/embedding.md (hunk 1 above)
Comment (changhiskhan) [reply docs/src/embedding.md:58]: added

File: docs/src/embedding.md (hunk 2 above)
Comment (changhiskhan) [reply docs/src/embedding.md:9]: it should. i'll add a note

File: docs/src/embedding.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+can be used by LanceDB as an embedding function.
+
+### HuggingFace example
+
+One popular free option would be to use the sentence-transformers library from HuggingFace.
+
+```python
+from sentence_transformers import SentenceTransformer
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply docs/src/embedding.md:17]: yes. added.

File: docs/src/embedding.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+For a given embedding function, the output will always have the same number of dimensions.
+
+## Creating an embedding function
+
+Any function that takes as input a batch (list) of data and outputs a batch (list) of embeddings
+can be used by LanceDB as an embedding function.
+
+### HuggingFace example
+
+One popular free option would be to use the sentence-transformers library from HuggingFace.
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply docs/src/embedding.md:14]: added

**Review by eddyxu — APPROVED** [review]

**Pull request merged**